docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+
+When we're using Git, we call this _cloning_ the teacher's repository (because we're making our own copy). After we've cloned the repository, we will have the same folder on our computer as the teacher created, committed, and pushed.
+
+TODO: Insert sequence diagram of teacher committing, pushing, and learners cloning.
+
+### Exercise (10 minutes)
+
+Code Your Future has already pushed an example repository, so you're going to try to clone it from Github onto your computer.
+
+1. Open up Github Desktop, and sign in if you're not already signed in.
+
+TODO: Insert screenshots for each step.
+
+2. You should see a button saying "Clone a Repository from the Internet" - click it. (If you don't see it, look in the menu at the top of the screen, open the "File" menu, and select the "Clone Repository" option).
+
+3. Now we need to tell Github Desktop where we want to repository we want to clone is. Choose the "URL" heading, and paste into the URL box: `https://github.com/illicitonion/cyf-demo-repo` (TODO: Replace with a CYF repo). The second box ("Local Path") is saying where it's going to clone it to on your computer. That's all Github Desktop is doing - copying a repository _from_ Github _to_ your computer. We don't need to change this box, but it can be useful to know where we can find our files later on.
+
+4. Press the Clone button. Github Desktop should soon show you a screen like this:
+
+TODO: Screenshot
+
+5. Click the "Open in Visual Studio Code" button (TODO: Explain to Linux users how to do this, because they probably don't have the button).
+
+6. Visual Studio code should open, and you should see an "Explorer" bar which says `CYF-DEMO-REPO` and under it `README.md`. If you click on where it says `README.md`, it should open the file. Congratulations, you have just cloned your first Git repository!
+
+## What did we just do?
+
+TODO: Diagram showing a teacher's copy, a copy on Github, and a copy on each of three students' laptops.
+
+Each of you just cloned a repository which CodeYourFuture created onto your computer, and opened up your copy of one of the files. This is the process you're going to follow to get your homework every week.
+
+## Let's do some homework (teacher-led demo)
+
+The file you opened, `README.md`, has a question in it. We're going to answer the question, and submit it as if it was our homework.
+
+We can edit the file in VS Code to answer the question, and save it like normal.
+
+After we've saved the file, if we open up Github Desktop, something interesting has changed. On the left, it now says "1 changed file", and on the right, it shows us what the change was. Things we've removed get a red background, and things we've added get a green background.
+
+This is a really useful way for us to check over our homework before submitting it. If we've accidentally deleted things, or changed things we didn't mean to, we can notice now, and un-do them by editing the file again.
```

Comment:
From @40thieves:

Big 👍 on emphasising inspecting the diff before committing.

I wonder if it would be useful to talk a bit about the "quirks" of reading diffs. E.g. editing a line is shown as removing the line and adding a new line with the change. My intuition is that it may come up in questions anyway.